Scripts/Get-UplinkDetails.ps1

```diff
@@ -0,0 +1,147 @@
+        $VMhostSwitch = $VMhost.NetworkInfo.VirtualSwitch
+
+        $objReport = @()
+        $VMhost| %{Get-View $_.ID} | 
+        %{ Get-View $_.ConfigManager.NetworkSystem} | 
+        %{ foreach($physnic in $_.NetworkInfo.Pnic){ 
+            
+            if($Type -eq "CDP"){
+                $obj = "" | Select-Object ClusterName,HostName,vmnic,PCI,MAC,VDS,vSwitch,CDP_Port,CDP_Device,CDP_Address
+            }
+            elseif($Type -eq "LLDP"){
+                $obj = "" | Select-Object ClusterName,HostName,vmnic,PCI,MAC,VDS,vSwitch,LLDP_Port,LLDP_Chassis,LLDP_SystemName
+                }
+                else{
+                    Throw "Invalide Type"
+                    }
+     
+            $pnicInfo = $_.QueryNetworkHint($physnic.Device) 
+            foreach($hint in $pnicInfo){ 
+                $obj.ClusterName = $VMhost.parent.name
+                $obj.HostName = $VMhost.name 
+                $obj.vmnic = $physnic.Device
+                $obj.PCI = $physnic.PCI
+                $obj.MAC = $physnic.Mac
+                if ($backing = ($VMhostProxySwitch | where {$_.Spec.Backing.PnicSpec.PnicDevice -eq $physnic.Device})) {
+                    $obj.VDS = $backing.DvsName
+                    } else {
+                        $obj.VDS = "-No Backing-"
+                        }
+                if ($backing = ($VMhostSwitch | where {$_.Nic -eq $physnic.Device})) {
+                    $obj.vSwitch = $backing.name
+                    } else {
+                        $obj.vSwitch = "-No Backing-"
+                        }
+                if($Type -eq "CDP"){
+                    if( $hint.ConnectedSwitchPort ) { 
+                        $obj.CDP_Port = $hint.ConnectedSwitchPort.PortId
+                        $obj.CDP_Device = $hint.ConnectedSwitchPort.DevId
+                        $obj.CDP_Address = $hint.ConnectedSwitchPort.Address  
+                        } else { 
+                            $obj.CDP_Port = "-No Info-" 
+                            $obj.CDP_Device = "-No Info-" 
+                            $obj.CDP_Address = "-No Info-" 
+                            }
+                        }
+                if($Type -eq "LLDP"){ 
+                    if( $hint.LldpInfo ) { 
+                        $obj.LLDP_Port = $hint.LldpInfo.PortId
+                        $obj.LLDP_Chassis = $hint.LldpInfo.ChassisId
+                        $obj.LLDP_SystemName = ($hint.LldpInfo.Parameter | where key -eq "System Name").Value
+                        } else { 
+                            $obj.LLDP_Port = "-No Info-" 
+                            $obj.LLDP_Chassis = "-No Info-" 
+                            $obj.LLDP_SystemName = "-No Info-" 
+                            }
+                        } 
+                
+
+            } 
+            $objReport += $obj 
+            } 
+        } 
+        $objReport 
+    } 
+  
+}
+
+Process {
+
+    $MyView = @()
```

Comment:
No need of having this array

`foreach () { Get-Info } | Sort-Object`

should be enough.
Do you really want sort to be part of this function. I'd recommend to return them without sorting and let the consumer decide how to sort them.

something like
`$VMHost | Foreach-Object {Write-Output (Get-Info $_)}`